src/scipp/spatial/transform/rotation.py

```diff
@@ -0,0 +1,130 @@
+from typing import Union, Sequence
+import numpy as _np
+
+from ... import units, Unit, UnitError, Variable, vectors
+from ..._scipp import core as _core_cpp
+
+
+def from_quat(*, value: Union[_np.ndarray, list]):
+    """
+    Creates a rotation-type variable from the provided quaternion coefficients.
+
+    The quaternion coefficients are provided in scalar-last order (x, y, z, w), where
+    x, y, z and w form the quaternion w + xi + yj + zk.
+
+    :param value: a numpy array or list with length 4, corresponding to the quaternion
+        coefficients (x*i, y*j, z*k, w)
+    """
+    return _core_cpp.rotations(dims=[], values=value)
+
+
+def from_quats(*, dims: Sequence[str], values: Union[_np.ndarray, list]):
+    """
+    Creates a rotation-type variable from the provided quaternion coefficients.
+
+    The quaternion coefficients are provided in scalar-last order (x, y, z, w), where
+    x, y, z and w form the quaternion w + xi + yj + zk.
+
+    :param values: a numpy array of numpy arrays corresponding to the quaternion
+        coefficients (w, x*i, y*j, z*k)
+    """
+    return _core_cpp.rotations(dims=dims, values=values)
+
+
+def from_matrix(*, value: Union[_np.ndarray, list]):
+    """
+    Create a single rotation transformation from a rotation matrix.
+
+    This requires ``scipy`` to be installed, as is wraps ``Rotation.from_matrix()``
+    from ``scipy.spatial.transform``.
+
+    :param value: a numpy 3x3 matrix, which will be interpreted as a rotation matrix.
+    """
+    from scipy.spatial.transform import Rotation as R
+    r = R.from_matrix(value)
+    return _core_cpp.rotations(dims=[], values=r.as_quat())
```

Comment:
So this turns a rotation matrix into a quaternion? Is it essential to have this now?